src/recon_buildblock/ML_estimate_component_based_normalisation.cxx

```diff
@@ -84,225 +127,274 @@ ML_estimate_component_based_normalisation(const std::string& out_filename_prefix
         }
     }
 
-  FanProjData model_fan_data;
-  FanProjData fan_data;
-  DetectorEfficiencies data_fan_sums(IndexRange2D(num_physical_rings, num_physical_detectors_per_ring));
-  DetectorEfficiencies efficiencies(IndexRange2D(num_physical_rings, num_physical_detectors_per_ring));
-
-  GeoData3D measured_geo_data(num_physical_axial_crystals_per_basic_unit,
-                              num_physical_transaxial_crystals_per_basic_unit / 2,
-                              num_physical_rings,
-                              num_physical_detectors_per_ring); // inputes have to be modified
-  GeoData3D norm_geo_data(num_physical_axial_crystals_per_basic_unit,
-                          num_physical_transaxial_crystals_per_basic_unit / 2,
-                          num_physical_rings,
-                          num_physical_detectors_per_ring); // inputes have to be modified
-
-  BlockData3D measured_block_data(num_axial_blocks, num_transaxial_blocks, num_axial_blocks - 1, num_transaxial_blocks - 1);
-  BlockData3D norm_block_data(num_axial_blocks, num_transaxial_blocks, num_axial_blocks - 1, num_transaxial_blocks - 1);
-
-  make_fan_data_remove_gaps(model_fan_data, model_data);
-  {
-    // next could be local if KL is not computed below
-    FanProjData measured_fan_data;
-    float threshold_for_KL;
-    // compute factors dependent on the data
+  // Setup the data structures given the PET scanner geometry
+  data_fan_sums = DetectorEfficiencies(IndexRange2D(num_physical_rings, num_physical_detectors_per_ring));
+  efficiencies_ptr = std::make_shared<DetectorEfficiencies>(IndexRange2D(num_physical_rings, num_physical_detectors_per_ring));
+
+  measured_geo_data = GeoData3D(num_physical_axial_crystals_per_basic_unit,
+                                num_physical_transaxial_crystals_per_basic_unit / 2,
+                                num_physical_rings,
+                                num_physical_detectors_per_ring); // inputes have to be modified
+  norm_geo_data_ptr = std::make_shared<GeoData3D>(num_physical_axial_crystals_per_basic_unit,
+                                                  num_physical_transaxial_crystals_per_basic_unit / 2,
+                                                  num_physical_rings,
+                                                  num_physical_detectors_per_ring); // inputes have to be modified
```

Comment:
2 `GeoData3D` objects? not great.

Reply:
They are not new https://github.com/UCL/STIR/blob/33a0106df6c796dbe718eea9c860565a41ff7c47/src/recon_buildblock/ML_estimate_component_based_normalisation.cxx#L92-L99

They are both used in `iterate_geo_norm(*norm_geo_data_ptr, measured_geo_data, fan_data);` Not sure how to simplify this

Reply:
ok fine. Sorry. Just correct "inputes" (although I don't know what those comments mean)

Reply:
It was probably a WIP note from before. I will remove them.